books/kestrel/number-theory/tonelli-shanks.lisp

```diff
@@ -185,25 +188,24 @@
 ;; that can't happen)
 
 ;; Future work: prove correctness; improve guards
-
+;; "p must be an odd prime"
 (define tonelli-shanks-sqrt ((n natp) (p natp) (z natp))
+  :guard (and (> p 2) (< z p) (rtl::primep p) (not (has-square-root? z p)))
   :short "Tonelli-Shanks modular square root."
   :long "Finds the square root of n modulo p.  p must be prime.
```

Comment:
Because there is xdoc, we should add a :parent and we should make the manual to make sure the parent works right.